community/kernel/src/test/java/org/neo4j/kernel/impl/core/RelationshipProxyTest.java

```diff
@@ -102,6 +108,30 @@ public RelationshipType answer( InvocationOnMock invocation ) throws Throwable
         }
     }
 
+    @Test
+    public void shouldPrintCypherEsqueRelationshipToString() throws Exception
+    {
+        // GIVEN
+        Node start, end;
+        RelationshipType type = RelationshipType.withName( "NICE" );
+        Relationship relationship;
+        try ( Transaction tx = db.beginTx() )
+        {
+            // GIVEN
+            start = db.createNode();
+            end = db.createNode();
+            relationship = start.createRelationshipTo( end, type );
+            tx.success();
+        }
+
+        // WHEN
+        String toString = relationship.toString();
+
+        // THEN
+        assertEquals( format( TO_STRING_FORMAT, start.getId(), type.name(), relationship.getId(), end.getId() ),
+                toString );
```

Comment:
It's generally bad form to replicate the product logic in the test code, since you can repeat bugs.

It would be better if you compared the resulting string to an expected string literal.